sdks/python/apache_beam/typehints/native_type_compatibility.py

```diff
@@ -198,6 +202,10 @@ def convert_to_beam_type(typ):
           arity=-1,
           beam_type=typehints.Tuple),
       _TypeMapEntry(match=_match_is_union, arity=-1, beam_type=typehints.Union),
+      _TypeMapEntry(
+          match=_match_issubclass(typing.Iterator),
```

Comment:
This will math all kinds of things, including strings and dicts, which is not what we want. (We may want to explicitly note that.)

Reply:
Line 170 effectively limits the scope to typing module types. Is that good enough?

Reply:
Ah, I see this is `Iterator`, not `Itererable`. I still think that changing what we do here should probably be a separate PR, or at least commit, as it's orthogonal to implementing `convert_to_typing_type` (or is there some reason you need it).

Reply:
I needed both directions of conversion between `typehints.IteratorTypeConstraint` and `typing.Iterator`.

Reply:
Sounds good. Thanks for bearing with me.

Reply:
PTAL. I don't see how I can implement bidirectional conversion of internal and external types without at least supporting all internal types. Otherwise we'll break existing users.

Reply:
Note that `GeneratorHint` has changed in my latest update. It now accepts 1 or 3 arguments in `__getitem__`.